python/test/unit/fem/test_assemble_domains.py

```diff
@@ -6,35 +6,163 @@
 """Unit tests for assembly over domains"""
 
 import numpy
+import pytest
+from petsc4py import PETSc
 
 import dolfin
 import ufl
 
 
-def test_basic_assembly_domains():
-    mesh = dolfin.generation.UnitSquareMesh(dolfin.MPI.comm_world, 5, 5)
+@pytest.fixture
+def mesh():
+    return dolfin.generation.UnitSquareMesh(dolfin.MPI.comm_world, 10, 10)
+
+
+def test_assembly_dx_domains(mesh):
+    V = dolfin.FunctionSpace(mesh, ("CG", 1))
+    u, v = dolfin.TrialFunction(V), dolfin.TestFunction(V)
+
+    marker = dolfin.MeshFunction("size_t", mesh, mesh.topology.dim, 0)
+    values = marker.array()
+    # Mark first, second and all other
+    # Their union is the whole domain
+    values[0] = 111
+    values[1] = 222
+    values[2:] = 333
+
+    dx = dolfin.Measure('dx', subdomain_data=marker, domain=mesh)
+
+    w = dolfin.Function(V)
+    w.vector().set(0.5)
+    w.vector().ghostUpdate(addv=PETSc.InsertMode.INSERT,
```

Comment:
This should be avoided. Set the 'local form' of the ghosted vector to avoid communication step.